test/e2e/storage/csi_objects.go

```diff
@@ -413,3 +468,92 @@ func deployGCEPDCSIDriver(
 	framework.ExpectNoError(err, "Failed to create DaemonSet: %v", nodeds.Name)
 
 }
+
+func printPodSecurityPolicies(client clientset.Interface, config framework.VolumeTestConfig) {
+	policys, err := client.PolicyV1beta1().PodSecurityPolicies().List(metav1.ListOptions{})
+	framework.ExpectNoError(err, "Failed to list PodSecurityPolicies")
+	framework.Logf("Got PodSecurityPolicies List")
+
+	for _, policy := range policys.Items {
+		framework.Logf("Got PodSecurityPolicy %s: %#v", policy.Name, policy.Spec)
+	}
+}
+
+// createGCESecrets downloads the GCP IAM Key for the default compute service account
+// and puts it in a secret for the GCE PD CSI Driver to consume
+func createGCESecrets(client clientset.Interface, config framework.VolumeTestConfig) {
+	var projectNumber string
+
+	gceCloud, err := framework.GetGCECloud()
+	framework.ExpectNoError(err, "Failed to get GCE Cloud")
+
+	var projectID string
+	if len(framework.TestContext.CloudConfig.ProjectID) != 0 {
+		projectID = framework.TestContext.CloudConfig.ProjectID
+	} else {
+		projectID = gceCloud.ProjectID()
+	}
+
+	_, err = strconv.ParseUint(projectID, 10, 64)
+	if err == nil {
+		// Project ID is a number
+		projectNumber = projectID
+
+	} else {
+		c, err := google.DefaultClient(context.Background(), cloudresourcemanager.CloudPlatformScope)
+		framework.ExpectNoError(err, "Failed to get Google Default Client")
+
+		cloudresourcemanagerService, err := cloudresourcemanager.New(c)
+		framework.ExpectNoError(err, "Failed to create new cloudresourcemanager")
+
+		resp, err := cloudresourcemanagerService.Projects.Get(projectID).Do()
+		framework.ExpectNoError(err, "Failed to get project %v from Cloud Resource Manager: %v", projectID)
+
+		projectNumber = strconv.FormatInt(resp.ProjectNumber, 10)
+	}
+
+	// Default Compute Engine service account
+	// [PROJECT_NUMBER]-compute@developer.gserviceaccount.com
+	serviceAccount := fmt.Sprintf("%v-compute@developer.gserviceaccount.com", projectNumber)
+
+	// Download Service Account JSON
+	saFile := "/tmp/cloud-sa.json"
+	args := []string{
+		"iam",
+		"service-accounts",
+		"keys",
+		"create",
+		saFile,
+		"--iam-account",
+		serviceAccount,
+		"--project",
+		projectID,
+	}
+	stdout, stderr, err := framework.RunCmd("gcloud", args...)
```

Comment:
In the CI this is generating keys which creates a lot of warning emails for us, and is leaving them behind, we at the very least probably need to extend the tooling to purge these keys after testing 😬

Reply:
Been thinking about this more (and getting more warnings), is it possible to identify the agent better? Right now I don't think we can differentiate between this tooling generating a key and someone trying to exfil a key via a naughty "test" or similar 🤔 I feel like end users might have similar concerns about these warnings.

Reply:
OK, we can at least tell that it is indeed the PR's service account creating this account, but it's unfortunate that we can't trivially tie it back to a job. maybe a very explicit log message so we can trivially track it back to the tests's logs?

Reply:
alright so i've updated it to shred the downloaded file and delete the key. Not sure what you mean by very explicit log message but I have this one:
`framework.Logf("Creating service account key for iam account %v in project %v at location %v", serviceAccount, projectID, saFile)`
let me know if you have something else in mind

Reply:
Can we use a pre-existing key in a secret or not run this in the default e2es? I'm concerned that the infra maintainers will have to either:
- filter out key download notifications (= bad we should be watching for these...)
- actually sift through a deluge of service account key download emails

We're getting an alert from GCP every time this runs, delayed by about one day.

Reply:
There's no way to "get" a service account key, you can only get the actual private key data on creation of the key so thats necessary.

A compromise here may be to have it still gated behind a `[Feature: _]` thing so that it doesn't run in CI by default, and turn it on when external pod identity goes beta/GA and we can use that.

@saad-ali, @msau42  do you have any opinions on how important this is to run in CI by default?

Reply:
@msau42 suggested maybe pre-creating and downloading one service key for each project and saving them somewhere "safe" to be used by this test. @BenTheElder is there a known/existing mechanism for doing something like this?

Reply:
we just discussed this offline, this is what I meant :-) we have keys that we use for CI resources, PR for anyone following along is https://github.com/kubernetes/test-infra/pull/9320